internal/legacy/api_test.go

```diff
@@ -329,61 +330,27 @@ func (s *APISuite) TestCharmPackageCharmInfo(c *gc.C) {
 }
 
 func (s *APISuite) TestSHA256Laziness(c *gc.C) {
-	updated := make(chan struct{}, 1)
-	// Patch updateEntitySHA256 so that we can know whether
-	// it has been called or not.
-	oldUpdate := *legacy.UpdateEntitySHA256
-	s.PatchValue(legacy.UpdateEntitySHA256, func(store *charmstore.Store, url *charm.Reference, sum256 string) {
-		oldUpdate(store, url, sum256)
-		updated <- struct{}{}
-	})
-
-	// Add a charm without a SHA256 hash.
-	wordpressURL, wordpress := s.addCharm(c, "wordpress", "cs:precise/wordpress-0")
-	s.store.DB.Entities().UpdateId(wordpressURL, bson.D{{
-		"$set", bson.D{{"blobhash256", ""}},
-	}})
-	sum256 := fileSHA256(c, wordpress.Path)
-
-	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
-		Handler:      s.srv,
-		URL:          "/charm-info?charms=" + wordpressURL.String(),
-		ExpectStatus: http.StatusOK,
-		ExpectBody: map[string]charm.InfoResponse{
-			wordpressURL.String(): {
-				CanonicalURL: wordpressURL.String(),
-				Sha256:       sum256,
-				Revision:     0,
-			},
-		},
-	})
-
-	select {
-	case <-updated:
-	case <-time.After(5 * time.Second):
-		c.Fatalf("timed out waiting for update")
-	}
-
-	// Try again - we should not update the SHA256 the second time.
-
-	httptesting.AssertJSONCall(c, httptesting.JSONCallParams{
-		Handler:      s.srv,
-		URL:          "/charm-info?charms=" + wordpressURL.String(),
-		ExpectStatus: http.StatusOK,
-		ExpectBody: map[string]charm.InfoResponse{
-			wordpressURL.String(): {
-				CanonicalURL: wordpressURL.String(),
-				Sha256:       sum256,
-				Revision:     0,
+	// TODO frankban: remove this test after updating entities in the
+	// production db with their SHA256 hash value. Entities are updated by
+	// running the cshash256 command.
+	id, ch := s.addCharm(c, "wordpress", "cs:precise/wordpress-0")
```

Comment:
Can you make this test use a charm with a user please? that way it won't break when the promulgation stuff merges.